src/main/java/org/apache/commons/io/build/AbstractOrigin.java

```diff
@@ -203,7 +452,7 @@ public CharSequenceOrigin(final CharSequence origin) {
         @Override
         public byte[] getByteArray() {
             // TODO Pass in a Charset? Consider if call sites actually need this.
-            return origin.toString().getBytes(Charset.defaultCharset());
+            return getCharSequence(null).toString().getBytes(Charset.defaultCharset());
```

Comment:
How is this change related to Channels? This is a giant PR already. Can it be focused on just the new feature? Or is this a bug fix?

Reply:
Fixed in https://github.com/apache/commons-io/pull/784/commits/24fff74dc1645e7b566fb167befe1ad167c1ba62